types/vote_test.go

```diff
@@ -53,6 +53,37 @@ func TestVoteSignable(t *testing.T) {
 	require.Equal(t, expected, signBytes, "Got unexpected sign bytes for Vote.")
 }
 
+func TestVoteSignableTestVectors(t *testing.T) {
+	//chainID := "test_chain_id"
+	tests := []struct {
+		vote    *Vote
+		chainID string
+		want    []byte
+	}{
+		// XXX: Here Height and Round are skipped. This probably will be cumbersome to to parse in the HSM:
+		{&Vote{}, "", []byte{0xb, 0x22, 0x9, 0x9, 0x0, 0x9, 0x6e, 0x88, 0xf1, 0xff, 0xff, 0xff}},
+		// with proper (fixed size) height and round:
+		{
+			&Vote{Height: 1, Round: 1},
+			"",
+			[]byte{
+				0x1d,                                   // total length
+				0x9,                                    // (field_number << 3) | wire_type
+				0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, // height
+				0x11,                                   // (field_number << 3) | wire_type
+				0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, // round
+				// remaining fields:
+				0x22, 0x9, 0x9, 0x0, 0x9, 0x6e, 0x88, 0xf1, 0xff, 0xff, 0xff},
```

Comment:
@jleni do this kind of tests tests suit your need for test-cases (as mentioned in https://github.com/tendermint/tendermint/issues/2545#issuecomment-427678845)?

Reply:
That is great. What about version number? 
It would be good that it is there too. So breaking changes can be signaled and the HSM can reject signing.

Reply:
btw, what checks would you recommend in HSM code? I am planning to check at least:

- total length == blob size
- version == whatever we agree on
- field_number == 0x09 (or the corresponding value)
- block number >= 0
- field_number == 0x11 (or the corresponding value)
- round number >= 0

Reply:
Thanks! Let me sync up with @ebuchman or someone from the tm-core team about the version field.
The checks looks reasonable! I'll think about it and let you know if there should be more checks.

Reply:
So, the story around the version field is that the version isn't fully integrated yet. I'll add the field to `CanocialVote` and could add it manually for a few test cases so it would unblock your work on the HSM parsing side. 
@ebuchman does it make sense to change the `Signable` interface to take a version before we integrating the "Block Protocol Version"?

Reply:
That is fine, what we have now is enough for me to continue. I can now use these offsets and layout, complete things and later adjust.
We will eventually need this field anyway.
Maybe it makes sense that you add it and set to zero for now, until the other code starts using it but I don't actually need this anyway.

Reply:
Did you notice that fields are skipped whenever they are the default value? Does that pose a problem for you? I think you checks would look slightly different, e.g. if the first field is missing field_number == 0x09 would be false but field_number == 0x11 would be true (in case the second field isn't zero too etc).

Reply:
Does that behaviour (skipping defaults) have an analog in proto3? Or is it amino specific?

Reply:
@Liamsi Yes, I am aware of that. No problem, I was planning to consider this case
@ebuchman AFAIK, it is the same in protobuf3

Reply:
> Does that behaviour (skipping defaults) have an analog in proto3? Or is it amino specific?

@jleni is right. It is the default behaviour in proto3, too. In amino this ought to be configurable via the `write_empty` tag, though: https://github.com/tendermint/go-amino/blob/faa6e731944e2b7b6a46ad202902851e8ce85bee/binary-encode.go#L404-L408
Surprisingly, this doesn't work here. If this isn't a problem while parsing, let's keep it that way. I'll investigate and open an issue in amino if necessary.